recipes/llvm-core/all/conanfile.py

```diff
@@ -1,429 +1,478 @@
+            "LLVM_ENABLE_ASSERTIONS": str(self.settings.build_type),
+            "LLVM_USE_PERF": self.options.use_perf,
+            "LLVM_ENABLE_LIBEDIT": self.options.get_safe("with_libedit", False),
+            "LLVM_ENABLE_Z3_SOLVER": self.options.with_z3,
+            "LLVM_ENABLE_FFI": self.options.with_ffi,
+            "LLVM_ENABLE_ZLIB": "FORCE_ON" if self.options.with_zlib else False,
+            "LLVM_ENABLE_LIBXML2": "FORCE_ON" if self.options.with_xml2 else False,
+            "LLVM_ENABLE_TERMINFO": self.options.with_terminfo
+        }
+        if self.options.targets != "all":
+            cmake_variables["LLVM_TARGETS_TO_BUILD"] = self.options.targets
 
-    def package(self):
-        self.copy('LICENSE.TXT', dst='licenses', src=self._source_subfolder)
-        lib_path = os.path.join(self.package_folder, 'lib')
+        self._apply_resource_limits(cmake_variables)
 
-        cmake = self._configure_cmake()
-        cmake.install()
+        # this capability is back-ported from LLVM 14.x
+        is_platform_ELF_based = self.settings.os in [
+            "Linux", "Android", "FreeBSD", "SunOS", "AIX", "Neutrino", "VxWorks"
+        ]
+        if is_platform_ELF_based:
+            self.output.info("ELF Platform Detected, optimizing memory usage during debug build linking.")
+            cmake_variables["LLVM_USE_SPLIT_DWARF"] = True
+
+        if is_msvc(self):
+            build_type = str(self.settings.build_type).upper()
+            cmake_variables[f"LLVM_USE_CRT_{build_type}"] = msvc_runtime_flag(self)
 
         if not self.options.shared:
-            for ext in ['.a', '.lib']:
-                lib = '**/lib/*LLVMTableGenGlobalISel{}'.format(ext)
-                self.copy(lib, dst='lib', keep_path=False)
-                lib = '*LLVMTableGenGlobalISel{}'.format(ext)
-                self.copy(lib, dst='lib', src='lib')
-
-            CMake(self).configure(args=['--graphviz=graph/llvm.dot'], source_dir='.', build_dir='.')
-            with chdir(self, 'graph'):
-                dot_text = load(self, 'llvm.dot').replace('\r\n', '\n')
-
-            dep_regex = re.compile(r'//\s(.+)\s->\s(.+)$', re.MULTILINE)
-            deps = re.findall(dep_regex, dot_text)
-
-            dummy_targets = defaultdict(list)
-            for target, dep in deps:
-                if not target.startswith('LLVM'):
-                    dummy_targets[target].append(dep)
-
-            cmake_targets = {
-                'libffi::libffi': 'ffi',
-                'ZLIB::ZLIB': 'z',
-                'Iconv::Iconv': 'iconv',
-                'LibXml2::LibXml2': 'xml2'
-            }
+            cmake_variables.update({
+                "DISABLE_LLVM_LINK_LLVM_DYLIB": True,
+                "LLVM_ENABLE_PIC": self.options.get_safe("fPIC", default=True)
+            })
 
-            components = defaultdict(list)
-            for lib, dep in deps:
-                if not lib.startswith('LLVM'):
```

Comment:
I think this needs to be kept. Without something like this, I’m seeing shared libraries (`libRemarks.so` and `libLTO.so`) in the produced package even when `shared=False`.

Reply:
Might also want to delete `*.so` and `*.so.*` from the package.

Reply:
[libRemarks](https://llvm.org/docs/Remarks.html) and [libLTO](https://llvm.org/docs/LinkTimeOptimization.html#liblto) are part of the LLVM infrastructure and not available as static libraries.

libLTO specifically is used by some out of tree linkers but not by LLVM itself - See https://discourse.llvm.org/t/how-to-use-dynamic-library-liblto-so/74541 for example.

Reply:
I understand these are not available statically. However I don’t think it’s correct to include shared libs in a static package. For example, if my intention of using a static package is to produce an executable that doesn’t depend on any shared libs for ease of deployment, now I’m forced to link to these 2 shared libs when I link to `llvm-core::llvm::core`, and the executable will not run in the target environment as intended.

I think we should delete them, and the users of the static package should simply be aware that these 2 components are not available.

Reply:
> `llvm-core::llvm-core` is a conan-generated target and would not ordinarily be available to consumers of LLVM

By “not available” do you just mean “not recommended”? Because I certainly have code using it, and it works fine (except for linking to the 2 shared libs).

Reply:
I do think that `LTO` and `Remarks` should probably be made separate components for the shared build, but IMHO this is something that can be added as a future improvement and shouldn't block this PR.